[Bug] The latest IPython does not support Python 3.8

# 🐛 Bug

Yesterday, IPython released version 8.13, which failed our tests afterwards. IPython 8.13+ supports Python 3.9+, whereas IPython 8.0-8.12 supports Python 3.8+. We use Python versions 3.8 to 3.10. We should shortly freeze the version of IPython.

## To reproduce
Tests are failed.

** Stack trace/error message **
```
=========================== short test summary info ============================
FAILED examples/bindingdb_deepdta/tutorial.ipynb:: - NBMAKE INTERNAL ERROR
FAILED examples/cmri_mpca/tutorial.ipynb:: - NBMAKE INTERNAL ERROR
FAILED examples/digits_dann/tutorial.ipynb:: - NBMAKE INTERNAL ERROR
FAILED examples/multisite_neuroimg_adapt/tutorial.ipynb:: - NBMAKE INTERNAL ERROR
FAILED examples/toy_domain_adaptation/tutorial.ipynb:: - NBMAKE INTERNAL ERROR
====== 5 failed, 244 passed, 2 skipped, 101 warnings in 765.14s (0:12:45) ======
```

## Expected Behaviour
Tests are passed.

--- a/docs/requirements.txt
+++ b/docs/requirements.txt
@@ -8,7 +8,7 @@ https://data.pyg.org/whl/torch-1.9.0%2Bcpu/torch_scatter-2.0.7-cp37-cp37m-linux_
 https://data.pyg.org/whl/torch-1.9.0%2Bcpu/torch_sparse-0.6.10-cp37-cp37m-linux_x86_64.whl
 https://download.pytorch.org/whl/cpu/torch-1.9.1%2Bcpu-cp37-cp37m-linux_x86_64.whl
 
-ipython<8.13
+ipython<=8.12
 
 nbsphinx
 nbsphinx-link

--- a/docs/requirements.txt
+++ b/docs/requirements.txt
@@ -8,7 +8,7 @@ https://data.pyg.org/whl/torch-1.9.0%2Bcpu/torch_scatter-2.0.7-cp37-cp37m-linux_
 https://data.pyg.org/whl/torch-1.9.0%2Bcpu/torch_sparse-0.6.10-cp37-cp37m-linux_x86_64.whl
 https://download.pytorch.org/whl/cpu/torch-1.9.1%2Bcpu-cp37-cp37m-linux_x86_64.whl
 
-ipython<9.0
+ipython<8.13
 
 nbsphinx
 nbsphinx-link

--- a/docs/requirements.txt
+++ b/docs/requirements.txt
@@ -8,7 +8,7 @@ https://data.pyg.org/whl/torch-1.9.0%2Bcpu/torch_scatter-2.0.7-cp37-cp37m-linux_
 https://data.pyg.org/whl/torch-1.9.0%2Bcpu/torch_sparse-0.6.10-cp37-cp37m-linux_x86_64.whl
 https://download.pytorch.org/whl/cpu/torch-1.9.1%2Bcpu-cp37-cp37m-linux_x86_64.whl
 
-ipython<=8.12
+ipython<8.13
 
 nbsphinx
 nbsphinx-link

--- a/docs/requirements.txt
+++ b/docs/requirements.txt
@@ -8,7 +8,7 @@ https://data.pyg.org/whl/torch-1.9.0%2Bcpu/torch_scatter-2.0.7-cp37-cp37m-linux_
 https://data.pyg.org/whl/torch-1.9.0%2Bcpu/torch_sparse-0.6.10-cp37-cp37m-linux_x86_64.whl
 https://download.pytorch.org/whl/cpu/torch-1.9.1%2Bcpu-cp37-cp37m-linux_x86_64.whl
 
-ipython<8.13
+ipython<=8.12.0
 
 nbsphinx
 nbsphinx-link

--- a/setup.py
+++ b/setup.py
@@ -47,7 +47,7 @@ image_requires = [
 # Dependencies for all examples and tutorials
 example_requires = [
     "ipykernel",
-    "ipython<8.13",  # IPython 8.13+ support Python 3.9+ only and IPython 8.0-8.12 supports Python 3.8+.
+    "ipython<=8.12.0",  # IPython 8.13+ support Python 3.9+ only and IPython 8.0-8.12 supports Python 3.8+.
     "matplotlib<=3.5.2",
     "nilearn",
     "Pillow",

--- a/setup.py
+++ b/setup.py
@@ -47,7 +47,7 @@ image_requires = [
 # Dependencies for all examples and tutorials
 example_requires = [
     "ipykernel",
-    "ipython",
+    "ipython<8.13",  # IPython 8.13+ support Python 3.9+ only and IPython 8.0-8.12 supports Python 3.8+.
     "matplotlib<=3.5.2",
     "nilearn",
     "Pillow",

--- a/setup.py
+++ b/setup.py
@@ -47,7 +47,7 @@ image_requires = [
 # Dependencies for all examples and tutorials
 example_requires = [
     "ipykernel",
-    "ipython<=8.12",  # IPython 8.13+ support Python 3.9+ only and IPython 8.0-8.12 supports Python 3.8+.
+    "ipython<8.13",  # IPython 8.13+ support Python 3.9+ only and IPython 8.0-8.12 supports Python 3.8+.
     "matplotlib<=3.5.2",
     "nilearn",
     "Pillow",

--- a/setup.py
+++ b/setup.py
@@ -47,7 +47,7 @@ image_requires = [
 # Dependencies for all examples and tutorials
 example_requires = [
     "ipykernel",
-    "ipython<8.13",  # IPython 8.13+ support Python 3.9+ only and IPython 8.0-8.12 supports Python 3.8+.
+    "ipython<=8.12",  # IPython 8.13+ support Python 3.9+ only and IPython 8.0-8.12 supports Python 3.8+.
     "matplotlib<=3.5.2",
     "nilearn",
     "Pillow",

